referenceframe/model.go

```diff
@@ -245,17 +245,49 @@ func (m *SimpleModel) DoF() []Limit {
 
 // MarshalJSON serializes a Model.
 func (m *SimpleModel) MarshalJSON() ([]byte, error) {
-	return json.Marshal(m.modelConfig)
+	type serialized struct {
```

Comment:
1. why is this struct anonymous?
2. why do we duplicate it in both the marshal & unmarshal functions?

Reply:
comments in this previous [PR](https://github.com/viamrobotics/rdk/pull/5144) indicated the team's aversion to multiple structs that convey the same information. This was my effort to continue to abide by that principle